src/tests/test_schedulers.cpp

```diff
@@ -357,6 +358,173 @@ SCENARIO("New thread scheduler depends on subscription")
+            return std::nullopt;
+        });
+
+        THEN("order of call-stack must be in order")
+        {
+            REQUIRE(task_future.wait_for(std::chrono::seconds{5}) == std::future_status::ready);
+            REQUIRE(task_future.valid());
+            REQUIRE(call_stack == std::vector<std::string>{
+                "task 1 starts",
+                "task 1 ends",
+                "task 2 starts",
+                "task 2 ends",
+                "task 3 runs",
+            });
+        }
+    }
+}
+
+SCENARIO("trampoline scheduler is thread local")
+{
+    WHEN("two threads are using the same trampoline scheduler")
+    {
+        auto scheduler = rpp::schedulers::trampoline{};
+        auto sub       = rpp::composite_subscription{};
+        auto worker    = scheduler.create_worker(sub);
+        rpp::subscription_guard guard{sub};
+
+        std::promise<std::vector<std::thread::id>> call_stack_1;
+        std::promise<std::vector<std::thread::id>> call_stack_2;
+        auto stack_future_1 = call_stack_1.get_future();
+        auto stack_future_2 = call_stack_2.get_future();
+
+        std::jthread thread_1([&] {
+            std::vector<std::thread::id> call_stack;
+
+            worker.schedule([&]() -> rpp::schedulers::optional_duration {
+                call_stack.push_back(std::this_thread::get_id());
+                // Scheduler inner schedulable
+                worker.schedule([&]() -> rpp::schedulers::optional_duration {
+                    call_stack.push_back(std::this_thread::get_id());
+                    // Scheduler another inner schedulable
+                    worker.schedule([&]() -> rpp::schedulers::optional_duration {
+                        call_stack.push_back(std::this_thread::get_id());
+                        call_stack_1.set_value(std::move(call_stack));
+                        return std::nullopt;
+                    });
+                    return std::nullopt;
+                });
+                return std::nullopt;
+            });
+        });
+        std::jthread thread_2([&] {
+            std::vector<std::thread::id> call_stack;
+
+            worker.schedule([&]() -> rpp::schedulers::optional_duration {
+                call_stack.push_back(std::this_thread::get_id());
+                // Scheduler inner schedulable
+                worker.schedule([&]() -> rpp::schedulers::optional_duration {
+                    call_stack.push_back(std::this_thread::get_id());
+                    // Scheduler another inner schedulable
+                    worker.schedule([&]() -> rpp::schedulers::optional_duration {
+                        call_stack.push_back(std::this_thread::get_id());
+                        call_stack_2.set_value(std::move(call_stack));
+                        return std::nullopt;
+                    });
+                    return std::nullopt;
+                });
+                return std::nullopt;
+            });
+        });
```

Comment:
This two code blocks looks tooooooo clooooose. Extract some function to generate such an blocks and call twice (for thread 1 and for thread 2)

Reply:
In which file would you suggest for such utility function?

Reply:
You can extract it into same file. Or even create lambda inside a test, it doesn't matter